### PR TITLE
Code mixing metric

### PR DESCRIPTION
**Pull request by arielge**
- Metric that utilizes generative LLM to estimate if a text is code-mixed
- Added language identification task + example card
- Format for Starling LM

### CODE REVIEW

**Review by elronbandel — APPROVED** [review]

**Review by elronbandel — COMMENTED** [review]
@arielge @eladven @yoavkatz I think the 7b model is crashing the machine on preparation test, what would you advise?

**Comment by arielge** [comment]
Probably the easiest is to not run `test_metric` if a gpu is not available. Let me see if that does the trick

**Comment by arielge** [comment]
OK so this is still not enough as it looks like just loading the model crashes the test... Any ideas?

File: src/unitxt/metrics.py [hunk 1]
```diff
@@ -3771,3 +3773,65 @@ def get_element_group(self, element, additional_input):
...
+                "templates.language_identification.simple",
+                "formats.models.starling",
+            ]
+        )
+
+    def before_process_multi_stream(self):
+        model_id = "Nexusflow/Starling-LM-7B-beta"
+        self.inference_model = HFPipelineBasedInferenceEngine(
+            model_name=model_id, max_new_tokens=1
+        )
```
Comment (elronbandel) [inline src/unitxt/metrics.py:3815]:
```suggestion
        if not hasattr(self, "inference_model") or self.inference_model is None:
            self.inference_model = HFPipelineBasedInferenceEngine(
                model_name=model_id, max_new_tokens=1
            )
```

**Review by elronbandel — CHANGES_REQUESTED** [review]
Make sure the model is instantiated only once (see my suggestion)

File: src/unitxt/metrics.py [hunk 2]
```diff
@@ -3771,3 +3773,65 @@ def get_element_group(self, element, additional_input):
...
+
+    Note that this metric is quite tailored to specific model-template combinations, as it relies on the assumption
+    that the model will complete the answer prefix "The text is written in ___" in a particular way.
+
+    """
+
+    main_score = "is_code_mixed"
+    reduction_map = {"mean": [main_score]}
+    prediction_type = "str"
+
```
Comment (elronbandel) [inline src/unitxt/metrics.py:3797]:
```suggestion
     inference_model: InferenceEngine = Field(defalut_factory=lambda:  HFPipelineBasedInferenceEngine(
            model_name="Nexusflow/Starling-LM-7B-beta", max_new_tokens=1, lazy_load=True
        )
```
Counting on https://github.com/IBM/unitxt/pull/980

**Review by elronbandel — COMMENTED** [review]
see the latest comments counting on https://github.com/IBM/unitxt/pull/980

**Review by elronbandel — APPROVED** [review]

**Pull request merged**